src/huggingface_hub/inference/_providers/hf_inference.py

```diff
@@ -96,13 +96,20 @@ def __init__(self):
     def _prepare_payload_as_dict(
         self, inputs: Any, parameters: Dict, provider_mapping_info: InferenceProviderMapping
     ) -> Optional[Dict]:
+        payload = filter_none(parameters)
         mapped_model = provider_mapping_info.provider_id
         payload_model = parameters.get("model") or mapped_model
 
         if payload_model is None or payload_model.startswith(("http://", "https://")):
             payload_model = "dummy"
 
-        return {**filter_none(parameters), "model": payload_model, "messages": inputs}
+        response_format = parameters.pop("response_format", None)
+        if response_format is not None and response_format["type"] == "json_schema":
```

Comment:
hmm, ok I start to understand why you had to have custom logic for each provider... :confused: